### PR TITLE
Deprecate HandleCurrentMessageLater

### PR DESCRIPTION
**Pull request by timbussmann**
relates to https://github.com/Particular/NServiceBus/pull/4991

### CODE REVIEW

File: nservicebus/upgrades/6to7/index.md [hunk 1]
```diff
@@ -253,3 +253,19 @@ NServiceBus Version 7 will only run installers when explicitly enabled via the `
 ## DistributionStrategy
 
 The `string SelectReceiver(string[] receiverAddresses)` signature has been removed from the `DistributionStrategy` base class. When writing a custom distribution strategy, implement the `string SelectDestination(DistributionContext context)` method instead which provides additional information usable for routing decisions. Receiver addresses can still be accessed via the `context.ReceiverAddresses` property.
+
+
+## HandleCurrentMessageLater
+
+The `IMessageHandlerContext.HandleCurrentMessageLater()` method has been deprecated. Due to the different transports, the actual behavior of this method could vary which is why it's usage should be replaced by more expressive APIs to provide a consistent behavior across all transports.
```
Comment (andreasohlund) [inline nservicebus/upgrades/6to7/index.md]: > Due to the different transports

Depending on transport

Comment (andreasohlund) [reply nservicebus/upgrades/6to7/index.md]: >vary which is why it's usage should be replaced by more expressive APIs t

Just remove?

File: nservicebus/upgrades/6to7/index.md [hunk 2]
```diff
@@ -253,3 +253,19 @@ NServiceBus Version 7 will only run installers when explicitly enabled via the `
...
+
+To handle the current message later and abort the current processing attempt, throw an exception in your message handler and let [recoverability](/nservicebus/recoverability) reschedule the message. Be aware of the following restrictions:
+* recoverability is only enabled when the transport is configured to use some sort of transactions (!= `TransactionMode.None`).
```
Comment (andreasohlund) [inline nservicebus/upgrades/6to7/index.md]: > recoverability is only enabled

Retries is only enabled, message will go to error q if no tx

File: nservicebus/upgrades/6to7/index.md (hunk 1 above)
Comment (timbussmann) [reply nservicebus/upgrades/6to7/index.md]: I removed the whole sentence

**Pull request merged**